app/router.go

```diff
@@ -136,6 +142,12 @@ func RegisterReportPostHandler(a Adder, router *mux.Router) {
 			return
 		}
 
+		// a.Add(..., buf) assumes buf is gzip'd msgpack
+		if !isMsgpack {
```

Comment:
Why not simply reject uncompressed msgpack reports? I think it's safe to do since we haven't done it in the past.